src/main/java/games/strategy/engine/lobby/server/db/UserController.java

```diff
@@ -29,19 +29,31 @@ public UserController() {
    * @return null if the user does not exist.
    */
   @Override
-  public HashedPassword getPassword(final String userName) {
-    final String sql = "select password from ta_users where username = ?";
-    try (final Connection con = connectionSupplier.get(); final PreparedStatement ps = con.prepareStatement(sql)) {
-      ps.setString(1, userName);
+  public HashedPassword getPassword(final String username) {
+    return getPasswordFromColumn(username, "coalesce(password, bcrypt_password)");
+  }
+
+  /**
+   * @return null if the user does not exist or the user has no legacy password stored.
+   */
+  @Override
```

Comment:
Javadocs are not necessary on overrides.  This comment should probably be moved to `UserDao.java`.